pkg/asset/installconfig/platformprovisioncheck.go

```diff
@@ -79,6 +79,9 @@ func (a *PlatformProvisionCheck) Generate(dependencies asset.Parents) error {
 	if !enabledCaps.Has(string(configv1.ClusterVersionCapabilityMachineAPI)) {
 		return errors.New("IPI requires MachineAPI capability")
 	}
+	if !enabledCaps.Has(string(configv1.ClusterVersionCapabilityIngress)) {
```

Comment:
Ingress capability should be always enabled in standalone ocp, not only on IPI, UPI should also be considered.
perhaps this check should be added into installconfig validation, https://github.com/openshift/installer/blob/master/pkg/types/validation/installconfig.go#L195?

Reply:
Right, that may be more appropriate place for the validation.

Reply:
In UPI, users can provide their own ingress and not use the ingress operator: https://docs.openshift.com/container-platform/4.15/installing/installing_aws/installing-aws-user-infra.html#installation-user-infra-generate-k8s-manifest-ignition_installing-aws-user-infra

I think it makes sense to only enforce this check for IPI

Reply:
From what I can see the users can set up [the network load balancer](https://docs.openshift.com/container-platform/4.15/installing/installing_aws/installing-aws-user-infra.html#installation-cloudformation-dns_installing-aws-user-infra) and [the dns records](https://docs.openshift.com/container-platform/4.15/installing/installing_aws/installing-aws-user-infra.html#installation-create-ingress-dns-records_installing-aws-user-infra). Both of them are required to send the network traffic to the ingresscontroller (router). The latter is still deployed (we can see this in the commands `oc -n openshift-ingress get service router-default`). The router is a part of the ingress capability. So if it would be disabled I don't understand where the setup NLB and DNS would send the traffic to.